### PR TITLE
Tukaiz develop

### PR DESCRIPTION
**Pull request by francirp**
I have customized the permissions but not yet started on the approval system. If you want to see it in action to have context let me know and I can walk you through it.

### CODE REVIEW

File: README.textile [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+1. Advanced permissions: LocomotiveCMS comes with three roles out of the box: Admin, Designer, and Author. For our purposes, we needed four roles: Super Admin, Site Admin, Advanced User, Beginner User. We customized the gem to work correctly for these four user types.
+2. Approval system: Edits to pages made by beginner users must be approved. This system includes page preview, versioning, and notification functionality.
+
+**Roles**
+* Super Admin: Full access to all sites and members
+* Site Admin: Full access to one site and that site's members that are not Super Admins
+* Advanced User: Can only access pages assigned by an Admin. Ability to manage all aspects of the pages they have access to.
+* Beginner User: Can only access pages assigned by an Admin. Ability to contribute to pages but contributions must be approved by an Advanced User, or Admin.
+
 h1. LocomotiveCMS
```
Comment (bhennes2) [inline README.textile:16]: Great stuff here.

File: app/models/locomotive/account.rb [hunk 2]
```diff
@@ -147,5 +148,23 @@ def remove_memberships!
...
+    end
+
+    def not_member_of_all_sites?
+      !member_of_all_sites?
+    end
+
+    def create_other_site_memberships
+      if admin? && not_member_of_all_sites?
+        Locomotive::Site.all.each do |site|
+          unless site.memberships.where(account_id: self._id).first
```
Comment (bhennes2) [inline app/models/locomotive/account.rb:162]: Can you use a find_or_create_by here?

File: app/controllers/locomotive/api/memberships_controller.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@ def show
       end
 
       def create
-        @membership.from_presenter(params[:membership].merge(role: 'author')) # force author by default
+        @membership.from_presenter(params[:membership].merge(role: Locomotive::Ability::DEFAULT_ROLE)) # force author by default
```
Comment (bhennes2) [inline app/controllers/locomotive/api/memberships_controller.rb:16]: Can we set the default role on the membership model?......in some type of callback?

**Comment by bhennes2** [comment]
Great stuff here. Simple and flexible for change.

File: app/models/locomotive/account.rb (hunk 2 above)
Comment (francirp) [inline app/models/locomotive/account.rb:162]: Right on - great call:

``` ruby
    def create_other_site_memberships
      if admin? && not_member_of_all_sites?
        Locomotive::Site.all.each do |site|
          site.memberships.find_or_create_by(account_id: self._id)
        end
      end
    end
```

File: app/controllers/locomotive/api/memberships_controller.rb (hunk 3 above)
Comment (francirp) [inline app/controllers/locomotive/api/memberships_controller.rb:16]: Ah good call. I did already set a default role to the membership model as follows:

``` ruby
    field :role, default: Locomotive::Ability::DEFAULT_ROLE
```

Which means this code should work for the create action in that controller:

``` ruby
@membership.from_presenter(params[:membership])
```

**Pull request merged**